src/goto-symex/symex_main.cpp

```diff
@@ -548,6 +548,56 @@ void goto_symext::run_intrinsic(
     // Already modelled in builtin_libs
     return;
   }
+  else if(has_prefix(symname, "c:@F@__ESBMC_init_var"))
+  {
+    assert(
+      func_call.operands.size() == 1 && "Wrong __ESBMC_init_var signature");
+    auto &ex_state = art.get_cur_state();
+    if(ex_state.cur_state->guard.is_false())
+      return;
+
+    // Get the argument
+    expr2tc arg0 = func_call.operands[0];
+    internal_deref_items.clear();
+    dereference2tc deref(get_empty_type(), arg0);
+    dereference(deref, dereferencet::INTERNAL);
+
+    for(const auto &item : internal_deref_items)
+    {
+      assert(
+        item.object->expr_id == expr2t::expr_ids::symbol_id &&
+        "__ESBMC_init_var only works for variables");
```

Comment:
There is not a hard requirement. This is the expected flow for it.

If a user has another use-case then it should be reported.

Reply:
In this case, `var` in `__ESBMC_init_var` seems misleading since it does only indirectly relate to variables in C:
```c
int x;
void *p = &x;
__ESBMC_init_var(p);
```
I get why it is taking arguments by pointer (because we don't have full control over the intrinsics built by clang). So maybe `__ESBMC_init_object` would be more accurate?

Reply:
I think `init_object` might cause even more confusion for users (although is more correct technically). Klee has the `make_symbolic(void*)` which could also work in this context.

Reply:
I've renamed into `init_object`